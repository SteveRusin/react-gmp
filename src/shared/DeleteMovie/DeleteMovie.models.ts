export interface DeleteMovieProps {
  onConfirm: () => void;
  onClose: () => void;
}
