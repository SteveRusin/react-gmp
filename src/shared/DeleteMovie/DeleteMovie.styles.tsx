import styled from 'styled-components';

import { FlexColumnContainer } from '@shared/FlexColumnContainer';

export const DeleteMovieWrapper = styled(FlexColumnContainer)`
  font-size: 1.5rem;
  max-height: 250px;
  width: 500px;
  padding: 30px 30px 60px;

  background-color: ${({ theme }) => theme.palette.accent};

  .confirm-button {
    font-size: 2rem;
    margin-left: auto;
  }
`;
