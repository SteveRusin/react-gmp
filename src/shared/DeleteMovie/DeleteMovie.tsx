import { FC } from 'react';

import { ButtonPrimary } from '@shared';
import { CloseModal, ModalTitle, ModalHeader } from '@shared/Modal';

import { DeleteMovieWrapper } from './DeleteMovie.styles';
import { DeleteMovieProps } from './DeleteMovie.models';

export const DeleteMovie: FC<DeleteMovieProps> = ({ onClose, onConfirm }) => {
  return (
    <DeleteMovieWrapper>
      <ModalHeader>
        <ModalTitle>DELETE MOVIE</ModalTitle>
        <CloseModal size='24' onClick={onClose} />
      </ModalHeader>
      <div>Are you sure you want to delete this movie?</div>
      <ButtonPrimary className="confirm-button" onClick={onConfirm}>CONFIRM</ButtonPrimary>
    </DeleteMovieWrapper>
  );
};
