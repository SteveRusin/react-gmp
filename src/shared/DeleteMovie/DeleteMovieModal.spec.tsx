import { Modal } from '@shared';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@theme';
import { DeleteMovieProps } from './DeleteMovie.models';
import { DeleteMovie } from './DeleteMovie';
import { render } from '@testing-library/react';

function renderComponent(props: DeleteMovieProps) {
  return (
    <ThemeProvider theme={defaultTheme}>
      <Modal>
        <DeleteMovie {...props} />
      </Modal>
    </ThemeProvider>
  );
}

describe('Delete movie componentComponent', () => {
  let portalRoot: HTMLDivElement;

  beforeEach(() => {
    portalRoot = document.createElement('div');
    portalRoot.setAttribute('id', 'portal-root');
    document.body.appendChild(portalRoot);
  });

  afterEach(() => {
    document.body.removeChild(portalRoot);
  });

  it('should renders correctly', () => {
    render(
      renderComponent({
        onClose: jest.fn(),
        onConfirm: jest.fn(),
      })
    );

    expect(portalRoot).toMatchSnapshot();
  });
});
