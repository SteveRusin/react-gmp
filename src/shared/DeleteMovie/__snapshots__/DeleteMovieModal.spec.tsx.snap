// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete movie componentComponent should renders correctly 1`] = `
<div
  id="portal-root"
>
  <div>
    <div
      class="sc-bqyKOL epsUjv"
    >
      <div
        class="sc-bdfBQB sc-kfzBvY dzdWDj dPYDJS"
      >
        <div
          class="sc-fodVek icvtoC"
        >
          <h2
            class="sc-kstqJO iHJPjr"
          >
            DELETE MOVIE
          </h2>
          <svg
            aria-hidden="true"
            class="StyledIconBase-ea9ulj-0 iuNRof sc-hBEYId fbvFpy"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.75.75 0 1 1 1.06 1.06L9.06 8l3.22 3.22a.75.75 0 1 1-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 0 1-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div>
          Are you sure you want to delete this movie?
        </div>
        <button
          class="sc-gsTEea sc-eCstlR jDOibV creYTS confirm-button"
        >
          CONFIRM
        </button>
      </div>
    </div>
  </div>
</div>
`;
